src/diff.rs

```diff
@@ -16,13 +19,132 @@ pub struct ChangedPublicItem {
     pub new: PublicItem,
 }
 
+impl ChangedPublicItem {
```

Comment:
If you are OK with it, I think it would be good to split this PR up where the first part (this PR) does the change that introduces tokens, and another PR that improves diffing. In other words, if we could remove any changes related to diffing from this PR, it would be good I think.

The reason is that we can have a focused discussion on the details of the core change that introduces tokens, without getting distracted by the diffing parts.

If you really want to do it all in one big PR that is perfectly fine with me too. Since you are the one that writes the code, you are in large parts the one that decides how we should work together to finalize this :)

Reply:
I do agree that it makes a more focussed discussion. But I needed to add the `ChangedToken` while working on the `Token` to make sure the chosen strategy made sense for for this use case as well. But if you want I will disentangle them. It depends if you are comfortable with such a large code change and feel that you can find all errors this way, or need it disentangled. I am willing to disentangle, but would like to know that my effort is 'needed' ;-).

Reply:
Yeah, it totally get that it made a lot of sense to develop both concepts together to make sure things worked together. That was the right move. But now we that we have concepts in place that we know works, we are at a point where we can choose one of two scenarios:

Scenario A
1. Have tokens merged after 1 time units from now (PR 1)
2. Have intra-item diffing merged after 2 time units from now (PR 2)

Scenario B
1. Have tokens and intra-item diffing merged after 2 time units from now (one big PR)

Out of these scenarios I prefer A.

... but again, we do this purely for fun, so I would not say we *"need"* scenario A. and if you find it boring to work in an incremental manner (which I totally understand) we will not do that. So basically: your call :) But *my* preference would be to do a split.

Reply:
Okay I will work on a split then. I do agree that it is more logical.